File: src/main/java/cc/sukazyo/cono/morny/bot/event/OnInlineQuery.java
package cc.sukazyo.cono.morny.bot.event;

import cc.sukazyo.cono.morny.MornyCoeur;
import cc.sukazyo.cono.morny.bot.api.EventListener;
import cc.sukazyo.cono.morny.util.EncryptUtils;
import com.pengrad.telegrambot.model.Update;
import com.pengrad.telegrambot.model.request.InlineQueryResultArticle;
import com.pengrad.telegrambot.model.request.InputTextMessageContent;
import com.pengrad.telegrambot.model.request.ParseMode;
import com.pengrad.telegrambot.request.AnswerInlineQuery;

import javax.annotation.Nonnull;

/**
 * telegram inlineQuery 功能的处理类，
 * 也是一个 InlineQueryManager（还没做）
 *
 * @since 0.4.1.3
 */
public class OnInlineQuery extends EventListener {
	
	/**
	 * @since 0.4.1.3
	 */
	@Override
	public boolean onInlineQuery (@Nonnull Update update) {
		MornyCoeur.extra().exec(new AnswerInlineQuery(update.inlineQuery().id(), new InlineQueryResultArticle[]{
				new InlineQueryResultArticle(
						EncryptUtils.encryptByMD5(update.inlineQuery().query()),
						"Raw Input",
						new InputTextMessageContent(update.inlineQuery().query()).parseMode(ParseMode.MarkdownV2)
				)
		}));
		return true;
	}
	
}

File: src/main/scala/cc/sukazyo/cono/morny/bot/event/OnInlineQuery.scala
package cc.sukazyo.cono.morny.bot.event

import cc.sukazyo.cono.morny.MornyCoeur
import cc.sukazyo.cono.morny.bot.api.EventListener
import cc.sukazyo.cono.morny.bot.query.InlineQueryUnit
import com.pengrad.telegrambot.model.Update
import com.pengrad.telegrambot.model.request.InlineQueryResult
import com.pengrad.telegrambot.request.AnswerInlineQuery

import scala.collection.mutable.ListBuffer
import scala.language.postfixOps
import scala.reflect.ClassTag

object OnInlineQuery extends EventListener {
	
	override def onInlineQuery (using update: Update): Boolean = {
		
		val results: List[InlineQueryUnit[_]] = MornyCoeur.queryManager query update
		
		var cacheTime = Int.MaxValue
		var isPersonal = InlineQueryUnit.defaults.IS_PERSONAL
		val resultAnswers = ListBuffer[InlineQueryResult[_]]()
		for (r <- results) {
			if (cacheTime > r.cacheTime) cacheTime = r.cacheTime
			if (r isPersonal) isPersonal = true
			resultAnswers += r.result
		}
		
		if (results isEmpty) return false
		
		MornyCoeur.extra exec AnswerInlineQuery(
			update.inlineQuery.id, resultAnswers toArray:_*
		).cacheTime(cacheTime).isPersonal(isPersonal)
		true
		
	}
	
}
